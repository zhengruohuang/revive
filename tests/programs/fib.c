#include "bare.h"
#include <stdio.h>

int main(int argc, char *argv[])
{
    int dest = 50;
    unsigned long long f1 = 1, f2 = 1;
    
    if (dest == 1 || dest == 2) {
        printf("Fib #%d = 1\n", dest);
        write_output(0, 1);
        return 0;
    }
    
    for (int i = 3; i <= dest; i++) {
        unsigned long long save_f2 = f2;
        f2 += f1;
        f1 = save_f2;
        
        if (!(i % 10)) {
            printf("Fib #%d = %llu\n", i, f2);
        }
    }
    
    printf("Fib #%d = %llu\n", dest, f2);
    write_output(0, (unsigned long)f2);
    return 0;
}

