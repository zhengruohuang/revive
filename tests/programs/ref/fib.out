Fib #10 = 55
Fib #20 = 6765
Fib #30 = 832040
Fib #40 = 102334155
Fib #50 = 12586269025
Fib #50 = 12586269025
